tests/providers/amazon/aws/hooks/test_s3.py

```diff
@@ -69,6 +69,11 @@ def test_parse_s3_url(self):
         parsed = S3Hook.parse_s3_url("s3://test/this/is/not/a-real-key.txt")
         assert parsed == ("test", "this/is/not/a-real-key.txt"), "Incorrect parsing of the s3 url"
 
+    def test_parse_s3_object_directory(self):
+        parsed = S3Hook.parse_s3_url("s3://test/this/is/not/a-real-s3-directory/")
+        assert parsed != ("test", "this/is/not/a-real-s3-directory"), "Incorrect parsing of the s3 url"
+        assert parsed == ("test", "this/is/not/a-real-s3-directory/"), "Incorrect parsing of the s3 url"
```

Comment:
```suggestion
        parsed = S3Hook.parse_s3_url("s3://test/this/is/not/a-real-s3-directory/")
        assert parsed == ("test", "this/is/not/a-real-s3-directory/"), "Incorrect parsing of the s3 url"
```
This is sufficient :) If the `==` fails, the other one will succeed automatically.